### PR TITLE
fix(pie-notification): DSW-2009 address design review round 3

### PR DESCRIPTION
**Pull request by raoufswe**
## Describe your changes (can list changeset entries if preferable)

- `supportingAction` should only render if `leadingAction` is passed
- The `isDismissible` control on Storybook should only be visible if `isCompact` is set to false
- Simplified the logic of rendering default and named slot icons
- Render a placeholder icon for stories

## Author Checklist (complete before requesting a review)
- [x] I have performed a self-review of my code
- [x] I have reviewed the `PIE Storybook`/`PIE Docs` PR preview
- [x] I have reviewed visual test updates properly before approving

### Reviewer 1 - Seb
- [x] I have reviewed the `PIE Storybook`/`PIE Docs` PR preview

### Reviewer 2
- [x] I have reviewed the `PIE Storybook`/`PIE Docs` PR preview

### CODE REVIEW

File: apps/pie-docs/src/components/notification/code/props.json [hunk 1]
```diff
@@ -127,7 +127,7 @@
                 "type": "code",
                 "item": ["{ text: string, ariaLabel?: string }"]
             },
-            "An object representing the supporting action of the notification.",
+            "An object representing the supporting action of the notification, Will only render if `leadingAction` is passed",
```
Comment (leksaBoiko) [inline apps/pie-docs/src/components/notification/code/props.json:130]: An object representing the supporting action of the notification, **will** only render if `leadingAction` is passed"

Comment (raoufswe) [reply apps/pie-docs/src/components/notification/code/props.json]: updated 💯

File: packages/components/pie-notification/src/index.ts [hunk 2]
```diff
@@ -144,87 +118,53 @@ export class PieNotification extends LitElement implements NotificationProps {
...
      * Called within the main render function.
      *
-     * @param {NonNullable<NotificationProps['variant']>} variant
-     * @param {boolean} hasExternalIcon
-     *
      * @private
      */
-    private renderIcon (variant: NonNullable<NotificationProps['variant']>, hasExternalIcon: boolean, hasIconClass: boolean): TemplateResult | typeof nothing {
+    private renderIcon (): TemplateResult | typeof nothing {
+        const hasExternalIcon = this._iconSlot.length > 0;
```
Comment (raoufswe) [inline packages/components/pie-notification/src/index.ts:163]: the value of `this._iconSlot.length` is reactive so we can listen to it directly to render the default icon for an icon if defined.

Comment (sebastienrocks) [reply packages/components/pie-notification/src/index.ts:163]: If this is something a bit out of the ordinary, is it worth a comment in the code?

Comment (raoufswe) [reply packages/components/pie-notification/src/index.ts:163]: I don't think it's out of the ordinary, it's probably just we didn't know it's the case. 

the lit docs already go over how to query slot elements [here](https://lit.dev/docs/api/decorators/#queryAssignedElements)

**Review by sebastienrocks — APPROVED** [review]

**Review by leksaBoiko — APPROVED** [review]

**Pull request merged**